doc/user_guide/transformations.rst

```diff
@@ -754,10 +754,9 @@ OpenCL functionality. It also relies upon the device acceleration support
 provided by the dl_esm_inf library (https://github.com/stfc/dl_esm_inf).
 
 
-.. note:: The generated OpenCL files follow the `--kernel-renaming` argument
-    conventions, but in addition to the `<modulename>` they also include the
-    `<kernelname>` as part of the filename in the format:
-    `<modulename>_<kernelname>_index.cl`
+.. note:: The generated OpenCL kernels are written in a file called
```

Comment:
I am not sure about FPGAs because the Xilinx can separate the compiling and linking stages, so it may be possible to create a single image from multiple OpenCL files. But this would need redoing how the current Makefiles for the Xilinx are done, so I think is better to stick for the single file for now.

> what difference would "recover the indexing convention of the kernel-renaming parameter" make to the resulting name?

I think kernel-renaming can switch between failing if the kernel file already exist (and a string comparison shows the file is not the same as the backend output string) and keep increasing the index until we find a file that does exist. The OpenCL just does the later. I could look at Config.get().kernel_naming and do something consistent.

But again I am not sure if this will need to be rewritten anyway when doing multiple invokes if we need to hardcode the files into the FortCL function or we will still merge all the kernels from all the invokes in a single file.

Reply:
It now compiles and runs the OpenCL and compiles the MPI+OpenCL when F90=mpif90